src/common/metadata.c

```diff
@@ -191,9 +191,15 @@ typedef struct dt_undo_metadata_t
   GList *after;       // list of key/value after
 } dt_undo_metadata_t;
 
-static gboolean _compare_metadata(gconstpointer a, gconstpointer b)
+static GList *_list_find_custom(GList *list, gpointer data)
```

Comment:
isn't the simple fix to only change:
`return g_strcmp0(a, b);`
to
`return a && b ? g_strcmp0(a, b) : FALSE;`

Maybe not, but reading the patch it seems that the proposed change is exactly equivalent exception for the check for null inside the loop below:

`   if(i->data ...)`

Reply:
I'm my thinking is wrong, please explain and add a comment otherwise someone reading the code will certainly convert back this part as it was before.

Reply:
this is a double list: for each metadata there the id and then the value.
As long as the metadata value is not equal to one of the metadata id that works fine.
But if somebody (like for the related issue) set the title value to 1, g_list_ find finds the key instead of the value. The the list pointer is wrong...
So in the _list_find_custom loop there is 2 g_list_next to pull the key record in the comparison.

Reply:
if(i->data ...
I've added this test on the pointer because g_strcmp0 is able to return 0 if the 2 pointers are NULL.

Reply:
Oh I see, by "double list" I understood doubly linked list but this is a list with 2 entries for each metadata and we need to skip one every two values. Now I see the double g_list_next()! Sorry, I was confused.